alf/environments/parallel_environment.py

```diff
@@ -141,6 +164,11 @@ def _reset(self):
         time_steps = [env.reset(self._blocking) for env in self._envs]
         if not self._blocking:
             time_steps = [promise() for promise in time_steps]
+        if self._spare_queue:
+            self._record_done(time_steps)
+            self._spare_promises = [
```

Comment:
need to run the existing spare_promises before changing to the new promises.

Reply:
Good catch.